### PR TITLE
[charts] Fix RTL legend

### LINKED ISSUE #12118: [charts] labels are incorrect in RTL
### Steps to reproduce

Link to live example:
We could just look at the PyChart demo
https://mui.com/x/react-charts/pie/

Steps:
1. Set the direction of the settings to Right to Left

### Current behavior

The legend labels cover the the legend marks.
They also stay on the right.
Screenshot_20240218-102809

### Expected behavior

The legend should be all the way on the left with the marks before the label text

### Context

_No response_

### Your environment

System:
    OS: Windows 11 10.0.22621
  Binaries:
    Node: 16.18.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.8.1 - ~\AppData\Roaming\npm\npm.CMD
    pnpm: 8.10.5 - ~\AppData\Roaming\npm\pnpm.CMD
  Browsers:
    Chrome: Not Found
    Edge: Chromium (121.0.2277.128)
  npmPackages:
    @emotion/react: ^11.11.3 => 11.11.3
    @emotion/styled: ^11.11.0 => 11.11.0
    @mui/base:  5.0.0-beta.32
    @mui/core-downloads-tracker:  5.15.6
    @mui/icons-material: ^5.15.6 => 5.15.6
    @mui/joy: ^5.0.0-alpha.84 => 5.0.0-beta.23
    @mui/lab: ^5.0.0-alpha.135 => 5.0.0-alpha.161
    @mui/material: ^5.15.6 => 5.15.6
    @mui/private-theming: ^5.13.7 => 5.15.6
    @mui/styled-engine:  5.15.6
    @mui/system:  5.15.6
    @mui/types:  7.2.13
    @mui/utils:  5.15.6
    @mui/x-charts: ^6.19.4 => 6.19.4
    @mui/x-date-pickers: ^6.11.2 => 6.19.0
    @types/react:  18.2.48
    react: ^18.2.0 => 18.2.0
    react-dom: ^18.2.0 => 18.2.0
    typescript:  4.9.5

**Search keywords**: RTL

### PR DESCRIPTION
**Pull request by alexfauquette**
Fix #12118
image

Maybe I should invert the order of square/text : put the square on the right of the text @samueltrevor do you have some insight on this aspect?

### CODE REVIEW

**Comment by samueltrevor** [comment]
Yes. You should invert the order of the square/text.

**Comment by alexfauquette** [comment]
This symetric aspect is somewhat satisfying :)

RTL/LTR

image

File: packages/x-charts/src/ChartsLegend/DefaultChartsLegend.tsx [hunk 1]
```diff
@@ -214,60 +215,49 @@ function DefaultChartsLegend(props: LegendRendererProps) {
...
+        ...item,
+        positionY:
+          item.positionY +
+          (direction === 'row'
+            ? rowMaxHeight[item.rowIndex] / 2 // Get the center of the entire row
+            : item.outerHeight / 2), // Get the center of the item
+      })),
+      totalWidthUsed,
+      totalHeightUsed,
+    ];
```
Comment (alexfauquette) [inline packages/x-charts/src/ChartsLegend/DefaultChartsLegend.tsx:229]: Before it was absolute position. Non I return the size of the legend, and position are relative to the inside of this box.

The `gapX`, `gapY` that are made to align the legend with the border of the chart are moved outside such that we can simply invert

`positionX` to `legendWidth - positionX` to revert line order without having to care about where the legend is in the the chart

**Review by LukasTy — APPROVED** [review]
Nice improvement! 💙

**Pull request merged**